conanfile.py

```diff
@@ -75,8 +85,46 @@ class Xrpl(ConanFile):
         'zlib/1.2.11',
     ]
 
+    def set_version(self):
+        BUILDINFO = f"src/ripple/protocol/impl/BuildInfo.cpp"
+        regex = r"versionString\s?=\s?\"(.*)\""
+        with open(BUILDINFO, 'r') as file:
+            while True:
+                if match := re.search(regex, file.readline()): break
+            self.version =  match.group(1)
+
     def requirements(self):
         if self.options.with_jemalloc:
             self.requires('jemalloc/5.2.1')
+        if self.options.reporting:
+            self.requires('cassandra-cpp-driver/2.15.3')
+            self.requires('libpq/13.6')
 
-    generators = 'CMakeToolchain', 'cmake_find_package'
+    def export_sources(self):
+        self.copy("*", excludes=["build", ".github"])
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        if not self.options.tests:
+            tc.variables["tests"] = False
+        tc.variables["reporting"] = self.options.reporting
+        tc.variables["unity"] = self.options.unity
+        tc.generate()
```

Comment:
If we use the `generate` method like this, then we should change the `generators` attribute on line 8. I would move that line to just before this method and change the value to just `CMakeDeps`.